app/Http/Livewire/ConferenceList.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+namespace App\Http\Livewire;
+
+use App\Models\Conference;
+use Carbon\CarbonImmutable;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+use Livewire\Component;
+
+class ConferenceList extends Component
+{
+    public $date;
+
+    public $year;
+
+    public $month;
+
+    public $filter = 'future';
+
+    public $sort = 'date';
+
+    public $search;
+
+    protected $queryString = [
+        'year',
+        'month',
+        'filter',
+        'sort',
+        'search',
+    ];
+
+    public function boot()
+    {
+        Builder::mixin($this->queryScopes());
```

Comment:
Uh,.. i dont think I've seen this pattern before. Can you share a link or something else to help me learn?!

Reply:
I noticed it while source diving the `Macroable` trait to experiment with defining macro methods on the calling class.  The `mixin` method accepts a class, which in this case I'm using an anonymous returned by the `queryScopes` method.  The catch is that the methods in the mixin class all need to return callables so each method is one level of indentation greater than they otherwise would be.

Reply:
Haha, that's awesome!